src/components/Admin/QuestionDisplay.js

```diff
@@ -0,0 +1,212 @@
+    this.loadSettings();
+  };
+
+  loadSettings = async () => {
+    this._initFirebase = true;
+
+    this.setState({ question: this.props.question, loading: false });
+  };
+
+  toggleEdit = () => {
+    this.setState({ editQuestion: !this.state.editQuestion });
+  };
+
+  toggleDelete = () => {
+    this.setState({ deleteQuestion: !this.state.deleteQuestion });
+  };
+
+  deleteQ = () => {
+    this.props.firebase
+      .question(this.props.question.id)
+      .delete()
+      .then(() => {
+        console.log("Question Deleted: ", this.props.question.id);
+        this.updateData();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  updateData = () => {
+    this.setState({ deleteQuestion: false, editQuestion: false });
+    this.props.updateFunc();
+  };
+
+  render() {
+    const {
+      loading,
+      error,
+      editQuestion,
+      question,
+      deleteQuestion,
+    } = this.state;
+
+    if (loading) return <Loader />;
+    if (error)
+      return (
+        <Container flexdirection="column">
+          <h1>{error}</h1>
+        </Container>
+      );
+
+    var hasIMG = false;
+    if (question.image !== "") hasIMG = true;
+
+    if (editQuestion) {
+      return (
+        <StyledCol md={4}>
+          <StyledDiv>
+            <h2>Edit Question</h2>
+
+            <StyledHr />
+            <EditQuestion
+              question={question}
+              uid={this.props.question.id}
+              updateFunc={this.updateData}
+            />
+            <StyledHr />
+
+            <Button onClick={this.toggleEdit}>Edit Data</Button>
+            <StyledHr />
+          </StyledDiv>
+        </StyledCol>
+      );
+    }
+
+    const Levels = () => {
+      return (
+        <Row>
+          <Col>
+            <h3> Level Scores </h3>
+
+            {Object.entries(question.scores).map((level) => (
+              <p key={level[0]}>
+                {" "}
+                {level[0]} | Average: {level[1].avrg} | Amount:{" "}
+                {level[1].amount}{" "}
+              </p>
+            ))}
+          </Col>
+        </Row>
+      );
+    };
+
+    const Delete = () => {
+      return (
+        <Fragment>
+          <Button onClick={this.deleteQ}>Are you sure?</Button>
+          <StyledHr />
+        </Fragment>
+      );
+    };
+
+    return (
+      <StyledCol md={4}>
+        <StyledDiv>
+          <h2> {question.name} </h2>
+
+          <StyledHr />
+          <h3> Answer </h3>
+          <p> {question.answer} </p>
+
+          <h3> Description </h3>
+          {hasIMG ? <img src={question.image} alt="Question" /> : <> </>}
+          <p> {question.description} </p>
+
+          <Levels />
+
+          <StyledHr />
+          <Button onClick={this.toggleEdit}>Edit Question</Button>
+          <br />
+          <br />
+          <Button onClick={this.toggleDelete}>Delete Question</Button>
+          <StyledHr />
+
+          {deleteQuestion ? <Delete /> : <> </>}
```

Comment:
Similar to earlier, this can be shortened to just `{deleteQuestion && <Delete />}`. However, instead of using state to keep track of a toggle to show a component, this can be handled by just having the Delete button call the delete function passed from the parent, and in that function call swal to confirm if the user really wants to delete.

Reply:
Shortened the if statement in commit 21706d9246057ef3da00d313a3a1ea132dedb6c1, will implement swal later today (not closing this issue)

Reply:
Resolved in commit 328335366bf582d828349cde311f9f15794bbca6